src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/Report.kt

```diff
@@ -67,38 +69,38 @@ class Report(
   var reportedBy: String,
   var reportedAt: LocalDateTime,
 
-  @ManyToOne(fetch = FetchType.EAGER, cascade = [CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH], optional = false)
+  @ManyToOne(fetch = FetchType.LAZY, cascade = [CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH], optional = false)
```

Comment:
This could be a real performance problem

Reply:
Don't we use most (if not all) the associated records of a report anyway? (e.g. when serialising to the `Report` DTO response class).

That means loading these associated records is actually slower as they would be loaded anyway lazily via additional SQL queries (the so called "N+1 queries problem")

Reply:
Trust me on this one.  Yes its true when you are loading the whole report but the problem is that it will ALWAYS load this data on ANY query you run.  So for you full DTO convert we can perform a batch fetch in the query - or use a `@NamedEntityGraph`